src/speech_to_text_finetune/finetune_whisper.py

```diff
@@ -25,75 +27,58 @@
     get_available_languages_in_cv,
 )
 
-hf_username = get_hf_username()
-dataset_id_cv = "mozilla-foundation/common_voice_17_0"
-model_id_whisper = "openai/whisper-tiny"
-test_language = "Greek"
-
-test_repo_name = "testing"  # None for default name, or set your own
-test_max_steps = 100
-push_to_hf = True
-make_repo_private = False
-
-
-def run_finetuning(
-    model_id: str,
-    dataset_id: str,
-    language: str,
-    repo_name: str | None,
-    max_steps: int = 2000,
-    private_hf_repo: bool = True,
-) -> Tuple[Dict, Dict]:
+
+def run_finetuning(config_path: str = "config.yaml") -> Tuple[Dict, Dict]:
     """
     Complete pipeline for preprocessing the Common Voice dataset and then finetuning a Whisper model on it.
 
     Args:
-        model_id (str): HF model id of a Whisper model used for finetuning
-        dataset_id (str): HF dataset id of a Common Voice dataset version, ideally from the mozilla-foundation repo
-        language (str): registered language string that is supported by the Common Voice dataset
-        repo_name (str): repo ID that will be used for storing artifacts both locally and on HF
-        max_steps (int): number of steps to run the training job, defaults to 2000
-        private_hf_repo (bool): flag whether to make the HF public (False) or private (True)
+        config_path (str): The filepath to a yaml file that follows the format defined in config.py
 
     Returns:
         Tuple[Dict, Dict]: evaluation metrics from the baseline and the finetuned models
     """
+    cfg = load_config(config_path)
 
-    languages_name_to_id = get_available_languages_in_cv(dataset_id)
-    language_id = languages_name_to_id[language]
+    hf_username = get_hf_username()
```

Comment:
I am not sure I follow why there is a `push_to_hf` argument but regardless of the value we set `hf_repo_name` and pass it to the `Seq2SeqTrainingArguments`

Reply:
its true that if you set `push_to_hf=False` then there is no point for `hf_repo_name`. My thinking was this:
- You want to be able to turn on/off uploading to HF. 
- You want to be able to set your own `hf_repo_name` in the config instead of getting an autogenerated name. e.g maybe you have been trying finetuning on your private repo and want to make a new public repo with a different name.

So instead of having a bunch of if/else for all these cases, I am just passing `hf_repo_name` to `Seq2SeqTrainingArguments` even when `push_to_hf=False` and thus it has no effect.

Do you have a better suggestion?

Reply:
No real suggestions, I guess I was confused with the Seq2SeqTraining API.